### ISSUE TITLE
[BOUNTY] Implement hashes as a datatype

### ISSUE THREAD

**Issue by Dishpit** [Dishpit/Omega#89]

**Comment by alamshafil** [comment]
Hello! I was wondering if this bounty was still active. I recently found this project and it seemed interesting to me.

I attempted to implement a *basic* hash table using the language, which means it is not a built-in type. Let me know what you think. Here is example code:

```omega
... Hashmap impl. code ...

var Hashmap = Hashmap();

var apples = ["a","p","p","l","e","s"]; // Spilt string into array since Omega does not support getting each char of string.
var dog = ["d","o","g"];

Hashmap.put(apples, ["Golden Delicious", "Granny Smith"]);
Hashmap.put(dog, "Golden Retriever");

Hashmap.remove(dog);

out Hashmap.get(apples); // -> ["Golden Delicious", "Granny Smith"]
out Hashmap.get(dog); // -> false
```

Would it be better to implement an hash table directly into the interpreter C code or add it to the Omega STL? I understand that a built-in option would allow for better syntax, while an Omega-implementation is bound to class rules (methods, etc).

Thanks!

**Comment by Dishpit** [comment]
Hey @alamshafil! Thanks for reaching out; yes, the bounty's still active.

The sample code could be a good solution, especially considering it can be included into the Omega STL to help it grow, though it'd only be a temporary solution since we're looking to get a built-in variant done directly in the interpreter, similar to how Omega's Arrays work.

In particular, "hashes" here would mean something vaguely similar to JavaScript's objects.

```omega
var my_hash = {"hash table!": "first hash key!"};
```

Getting and setting would utilize the dot notation, just like how Omega's classes use the dot notation to add new methods to existing classes, and would require modification of the dot notation so it works for more than just Omega's classes
```omega
my_hash.dog = "Golden Retriever";
my_hash.apples = ["Golden Delicious", "Granny Smith"];
```

Depending on the key structure, users would need to utilize bracket notation to receive data. This would require some modification to the current bracket notation so it works for more than just Omega's arrays
```omega
out my_hash["hash table!"]; // -> "first hash hey!"
out my_hash.apples; // -> ["Golden Delicious", "Granny Smith"];
out my_hash.apples[0]; // -> "Golden Delicious"
```

In keeping with the [current Omega built-in function syntax](https://omegalang.com/docs/language-reference/builtin-functions#appendx-y) (clock(), append(), last(), etc), Hashmap.remove() would be a function that works like this:
```omega
out my_hash; // -> {"hash table!": "first hash key!", dog: "Golden Retriever", apples: ["Golden Delicious", "Granny Smith"]}
remove(my_hash, dog);
out my_hash; // -> {"hash table!": "first hash key!", apples: ["Golden Delicious", "Granny Smith"]}
```

**Comment by alamshafil** [comment]
Okay, I have implemented it in PR #120!

**Comment by Dishpit** [comment]
> Okay, I have implemented it in PR #120!

Great, I'll go over and review it momentarily.

Would you mind sending me a DM on twitter to exchange payment info?
